src/dev/flang/util/ANY.java

```diff
@@ -68,7 +74,24 @@ private static String origin(){
       {
         return "Unknown origin.";
       }
-    return st[2].getClassName() + ":" + st[2].getMethodName() + ":" + st[2].getLineNumber();
+    try (Stream<String> lines = Files.lines(Path.of(System.getProperty(FuzionConstants.FUZION_HOME_PROPERTY) + "/../src/" + st[2].getClassName().replaceAll("\\.", "/") + ".java")))
+      {
+        var previous = new AtomicReference<>("");
+        return st[2].getFileName() + ":" + st[2].getLineNumber() + Terminal.BLUE + " \""
+          + lines.skip(st[2].getLineNumber()-1).map(str -> str.trim())
+            .takeWhile(str -> {
+              var result = !previous.get().endsWith(");");
+              previous.set(str);
+              return result;
+            })
+            .limit(7)
+            .collect(Collectors.joining(" ")) + "\""
+          + Terminal.REGULAR_COLOR;
```

Comment:
There are no threads involved, so no need to use `Atomic`. I guess this is only to avoid the `java` limitation that locals cannot be modified in a lambda. What you could on instead is use an array: 
```
  var previous = new String[] { "" };
  ...
    previous[0] = ...
  ...
```
or, even better, try to avoid the side-effect in the lambda by doing something like shown below:
```suggestion
        var l = lines.skip(st[2].getLineNumber()-1).map(str -> str.trim());           
        return st[2].getFileName() + ":" + st[2].getLineNumber() + Terminal.BLUE + " \""
          + l.trim(Math.min(l.takeWhile(s->!s.endsWIth(");")).count()+1,7)
            .collect(Collectors.joining(" ")) + "\""
          + Terminal.REGULAR_COLOR;
```